stdlib/public/core/UnicodeScalarProperties.swift

```diff
@@ -0,0 +1,1402 @@
+  public var lowercaseMapping: String {
+    return _applyMapping(__swift_stdlib_u_strToLower)
+  }
+
+  /// The titlecase mapping of the scalar.
+  ///
+  /// This property is a `String`, not a `Unicode.Scalar` or `Character`,
+  /// because some mappings may transform a scalar into multiple scalars or
+  /// graphemes. For example, the ligature "ﬁ" (U+FB01 LATIN SMALL LIGATURE FI)
+  /// becomes "Fi" (U+0046 LATIN CAPITAL LETTER F, U+0069 LATIN SMALL LETTER I)
+  /// when converted to titlecase.
+  ///
+  /// This function corresponds to the `Titlecase_Mapping` property in the
+  /// [Unicode Standard](http://www.unicode.org/versions/latest/).
+  ///
+  /// - Returns: The titlecase mapping of the scalar.
+  public var titlecaseMapping: String {
+    return _applyMapping { ptr, cap, src, len, locale, err in
+      return __swift_stdlib_u_strToTitle(ptr, cap, src, len, nil, locale, err)
+    }
+  }
+
+  /// The uppercase mapping of the scalar.
+  ///
+  /// This property is a `String`, not a `Unicode.Scalar` or `Character`,
+  /// because some mappings may transform a scalar into multiple scalars or
+  /// graphemes. For example, the German letter "ß" (U+00DF LATIN SMALL LETTER
+  /// SHARP S) becomes "SS" (U+0053 LATIN CAPITAL LETTER S, U+0053 LATIN CAPITAL
+  /// LETTER S) when converted to uppercase.
+  ///
+  /// This function corresponds to the `Uppercase_Mapping` property in the
+  /// [Unicode Standard](http://www.unicode.org/versions/latest/).
+  ///
+  /// - Returns: The uppercase mapping of the scalar.
+  public var uppercaseMapping: String {
+    return _applyMapping(__swift_stdlib_u_strToUpper)
+  }
+}
+
+extension Unicode {
+
+  /// A version of the Unicode Standard represented by its `major.minor`
+  /// components.
+  public typealias Version = (major: Int, minor: Int)
+}
+
+extension Unicode.Scalar.Properties {
+
+  /// The earliest version of the Unicode Standard in which the scalar was
+  /// assigned.
+  ///
+  /// This value will be nil for code points that have not yet been assigned.
+  ///
+  /// This property corresponds to the `Age` property in the
+  /// [Unicode Standard](http://www.unicode.org/versions/latest/).
+  public var age: Unicode.Version? {
+    var versionInfo: __swift_stdlib_UVersionInfo = (0, 0, 0, 0)
+    withUnsafeMutablePointer(to: &versionInfo) { tuplePtr in
+      tuplePtr.withMemoryRebound(to: UInt8.self, capacity: 4) {
+        versionInfoPtr in
+        __swift_stdlib_u_charAge(_value, versionInfoPtr)
+      }
+    }
+    guard versionInfo.0 != 0 else { return nil }
+    return (major: Int(versionInfo.0), minor: Int(versionInfo.1))
+  }
+}
+
+extension Unicode {
+
+  /// The most general classification of a Unicode scalar.
+  ///
+  /// The general category of a scalar is its "first-order, most usual
+  /// categorization". It does not attempt to cover multiple uses of some
+  /// scalars, such as the use of letters to represent Roman numerals.
+  public enum GeneralCategory {
```

Comment:
I thought the discussion agreed on making that enum RawRepresentable (which would make it useful for building regular expressions for example)?

https://forums.swift.org/t/se-0211-add-unicode-properties-to-unicode-scalar/12121/6

Reply:
Since the proposal was accepted by the core team with the only revision being to the `numericValue` property, I didn't make any changes from the original proposal other than the one that was requested (and to remove a couple ICU-only properties that accidentally got into the list).

@milseman What do you think? Did the core team have any feelings about that specifically, or was it just an idea that just got lost in the discussion thread?

Reply:
Btw I don't think it would be _that_ useful to be able to do in the direction of `init(rawValue: String)` using `"Lu"` for example, but would _definitely_ be useful to go in the other direction (`. uppercaseLetter` => `"Lu"`). So only implementing `var description: String` for that purpose (instead of making it `RawRepresentable where RawValue: StringProtocol`) could also be enough

Reply:
There's many ways of describing it. There's the short name, the long name, how it might appear in various regex conventions, etc. I don't have any strong opinions here, other than `description` and `debugDescription` probably should choose user-readable names over  a particular regex convention.

Long term, Swift built in regexes and/or regex packages of various flavors could provide their `Custom[PCRE|POSIX|...]RegexConvertible` protocols. The would probably produces types that themselves could be `CustomStringConvertible`, producing e.g. `"\\p{Lu}"` or `"<:Lu>"`, etc.